search/tests/test_views.py

```diff
@@ -447,3 +447,84 @@ def test_search_from_url(self):
         searcher = SearchEngine.get_search_engine(TEST_INDEX_NAME)
         with self.assertRaises(StandardError):
             searcher.index("courseware_content", [{"id": "FAKE_ID_3", "content": {"text": "Here comes the sun"}}])
+
+
+@override_settings(SEARCH_ENGINE="search.tests.utils.ForceRefreshElasticSearchEngine")
+@override_settings(ELASTIC_FIELD_MAPPINGS={"start_date": {"type": "date"}})
+@override_settings(COURSEWARE_INDEX_NAME=TEST_INDEX_NAME)
+class ElasticSearchUrlTest(TestCase, SearcherMixin):
+    """Elastic-specific tests"""
+    def setUp(self):
+        self.searcher.index(
+            "courseware_content",
+            [
+                {
+                    "course": "ABC/DEF/GHI",
+                    "id": "FAKE_ID_1",
+                    "content": {
+                        "text": "It seems like k-means clustering would work in this context."
+                    },
+                    "test_date": datetime(2015, 1, 1),
+                    "test_string": "ABC, It's easy as 123"
+                }
+            ]
+        )
+        self.searcher.index(
+            "courseware_content",
+            [
+                {
+                    "course": "ABC/DEF/GHI",
+                    "id": "FAKE_ID_2",
+                    "content": {
+                        "text": "It looks like k-means clustering could work in this context."
+                    }
+                }
+            ]
+        )
+        self.searcher.index(
+            "courseware_content",
+            [
+                {
+                    "course": "ABC/DEF/GHI",
+                    "id": "FAKE_ID_3",
+                    "content": {
+                        "text": "It looks like k means something different in this context."
+                    }
+                }
+            ]
+        )
+
+    def test_quoted_phrase(self):
+        code, results = post_request({"search_string": "\"in this context\""})
+        self.assertTrue(199 < code < 300)
```

Comment:
nit: Why not just check for code == 200?

Reply:
@pomegranited my two cents - because there are a bunch of other "success" codes (like 204 NO CONTENT, or 201 CREATED) that might be used to indicate successful response. So it sort of makes sense as it makes the test more robust against changes in response status (i.e. if there are nothing to return someone might think it's a good idea to return 204, not 200 and in this case this test won't break).

Reply:
Thanks for your input @e-kolpakov !  I'm happy with this code as-is.

Reply:
Yeah, I was following the other status checks in the tests. Don't see why a synchronous search would return anything other than 200, but...

Reply:
With the current test setup, I believe its obvious that this would return 200. If someone makes changes in `setUp` and the expected search does not return anything with `204`, I will be clearer/cleaner to fail the test to catch the failure due to the changes. 
Thoughts?